docs/coding-guidelines/project-guidelines.md

```diff
@@ -80,37 +76,64 @@ When building an individual project the `BuildTargetFramework` and `TargetOS` wi
 Example:
 ```
 <PropertyGroup>
-  <TargetFrameworks>netstandard2.0;netstandard2.1</TargetFrameworks>
+  <TargetFrameworks>$(NetCoreAppCurrent);netstandard2.0;$(TargetFrameworkMinimum)</TargetFrameworks>
 </PropertyGroup>
+<ItemGroup Condition="'$(TargetFramework)' == '$(NetCoreAppCurrent)'">...</ItemGroup>
 <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">...</ItemGroup>
+<ItemGroup Condition="'$(TargetFramework)' == '$(TargetFrameworkMinimum)'">...</ItemGroup>
 ```
-2. Use a StartsWith when you want to test for multiple .NETStandard or .NETFramework versions.
+3. Use the `TargetPlatformIdentifier` property to condition on a .NETCoreApp platform specific target framework. Note that .NETStandard and .NETFramework target frameworks can't be platform specific.
 Example:
 ```
 <PropertyGroup>
-  <TargetFrameworks>netstandard2.0;netstandard2.1</TargetFrameworks>
+  <TargetFrameworks>$(NetCoreAppCurrent)-windows;$(NetCoreAppCurrent)-OSX;$(NetCoreAppCurrent)</TargetFrameworks>
+</PropertyGroup>
+<ItemGroup Condition="'$(TargetPlatformIdentifier)' == 'windows'">...</ItemGroup>
+<ItemGroup Condition="'$(TargetPlatformIdentifier)' == 'OSX'">...</ItemGroup>
+```
+Important: In contrast to the old `Targets*` checks, `TargetPlatformIdentifier` conditions apply to a single tfm only, inheritance between target frameworks can't be expressed. See the example below for Unix:
+```
+<PropertyGroup>
+  <TargetFrameworks>$(NetCoreAppCurrent)-Unix;$(NetCoreAppCurrent)-Linux;$(NetCoreAppCurrent)-android;$(NetCoreAppCurrent)-windows</TargetFrameworks>
 </PropertyGroup>
-<ItemGroup Condition="$(TargetFramework.StartsWith('netstandard'))>...</ItemGroup>
+<ItemGroup Condition="'$(TargetPlatformIdentifier)' == 'Unix' or '$(TargetPlatformIdentifier)' == 'Linux' or '$(TargetPlatformIdentifier)' == 'android'">...</ItemGroup>
+<!-- Negations make such conditions easier to write and read. -->
+<ItemGroup Condition="'$(TargetPlatformIdentifier)' != 'windows'">...</ItemGroup>
 ```
-3. Use a StartsWith if the TargetFramework is overloaded with the OS portion.
+4. Set the `TargetPlatformIdentifier` property in the project to be able to condition on it in properties in the project file.
+That is necessary as the SDK sets the `TargetPlatformIdentifier` in a .targets file after the project is evaluated. Because of that, the property isn't available during the project's evaluation and must be set manually.
 Example:
 ```
 <PropertyGroup>
-  <TargetFrameworks>netstandard2.0-windows;netstandard2.0-Unix</TargetFrameworks>
+  <TargetFrameworks>$(NetCoreAppCurrent)-windows;$(NetCoreAppCurrent)-android</TargetFrameworks>
+</PropertyGroup>
+<!-- DesignTimeBuild requires all the TargetFramework Derived Properties to not be present in the first property group. -->
+<PropertyGroup>
+  <TargetPlatformIdentifier>$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)'))</TargetPlatformIdentifier>
```

Comment:
could we move this to a .props file that we import here so that we don't need to have it in every project?

Reply:
Unfortunately not as it relies on the TargetFramework property. See https://github.com/dotnet/runtime/pull/64500#discussion_r799879098. @joperezr already asked the same question :)

Reply:
@ViktorHofer my suggestion was a little different, I meant 

```
<Import Project="$(RepositoryEngineeringDir)tpi.props"/>
<!-- DesignTimeBuild requires all the TargetFramework Derived Properties to not be present in the first property group. -->
<PropertyGroup>
   <DefineConstants Condition="'$(TargetPlatformIdentifier)' == 'android'">$(DefineConstants);ANDROID_USE_BUFFER</DefineConstants>
</PropertyGroup>
```

Reply:
You would trade one line to assign the TPI property with one line to import the props file in which you then again find the property assignment. I don't think we would need any additional helper properties.
An import is more expensive than a property assignment and our project evaluations are already unusually high so I would rather not have such an import unless we think that file would be a good place for other additional helper properties that are dependent on the TargetFramework property.